src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/HiddenItem.java

```diff
@@ -0,0 +1,24 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items;
+
+import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup;
+import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem;
+import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
+import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;
+import io.github.thebusybiscuit.slimefun4.core.attributes.NotConfigurable;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.bukkit.inventory.ItemStack;
+
+/**
+ * The {@link HiddenItem} is a {@link NotConfigurable} {@link SlimefunItem}
+ * that is hidden from the Slimefun guide.
+ */
+public class HiddenItem extends SlimefunItem implements NotConfigurable {
+
+    @ParametersAreNonnullByDefault
+    public HiddenItem(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(itemGroup, item, recipeType, recipe);
+        this.setHidden(true);
```

Comment:
couldnt we default the recipe to an empty `ItemStack[]` (or 9 nulls) meaning the ctor would get this declaration:
`public HiddenItem(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType)`
since this will always be hidden?

That way we dont have to initialize them in `SlimefunItemSetup` with a lot of nulls

Reply:
well right now it has the empty flask as an ingredient, i guess it would be better to have it just be nulls, would that be better?

Reply:
well the thing is the recipe doesnt matter right?
so we could make the constructor of HiddenItem something like:

```java
@ParametersAreNonnullByDefault
    public HiddenItem(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType) {
        super(itemGroup, item, recipeType, new ItemStack[] {});
        this.setHidden(true);
```

this means that if we then create an item somewhere else, we dont even have to initalize a recipe, because its not used anyway

Reply:
> the thing is the recipe doesnt matter right?

I don't really agree. It does matter as we have a new RecipeType that shows players what item should be right-clicked with to obtain the result item (in this case the filled knowledge flask).
This only makes sense if we do show the recipe item (in this case the empty flask)

Reply:
fair enough, but in that case should we just have a singular ItemStack then, instead of an array?

Reply:
```java
@ParametersAreNonnullByDefault
    public HiddenItem(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack recipe) {
        super(itemGroup, item, recipeType, new ItemStack[] { recipe, null, null, null, null, null, null, null, null });
        this.setHidden(true);
```

Reply:
What about `recipeItem` instead?

Reply:
That wouldn't work actually. Hidden items are not necessarily of RecipeType INTERACT. This would limit hidden items to only be able to have a single ingredient, which I'm not sure is the intention.

Reply:
Ah, good point. We should then leave it as it was before this, I don't think that the 8 nulls would be a huge problem anyway. We have a bunch of them in that file already, I think it's okay.